Add Support for No meeting days in Course creation

Formasitchijoh opened the pull request:
## What this PR does
This PR solve issue #6291 , by adding support for courses with timelines but without set meeting days, this is achieved by adding a field(**`no_meeting_days`**)  to the course table that keeps track of the status of the course if it has set meeting days or not and using it to compute the timeline appropriately. 

## Screenshots
After
## Course with no meeting days sets
https://www.loom.com/share/e0d9517577d34afe8535d4c1c62c228c?sid=88da0d2e-cbaa-4a4d-b9f6-e7ee4ae7664b

## Course with meeting days set 
https://www.loom.com/share/566c2eab6c0c4ea394e90773b276b8cc?sid=887877d2-c000-4dbb-a3e6-56923459f584

Formasitchijoh committed:
feat: add support for courses with timelines but without set meeting days

Formasitchijoh committed:
feat: add no_meetings_days as a flag to the course model

Formasitchijoh committed:
fix: removed unused content

Formasitchijoh committed:
fix:undefined method no_meeting_days in build failure

Formasitchijoh committed:
fix: updated course_creation test to include missing module

Formasitchijoh committed:
remove meeting days on summary course page display

Formasitchijoh commented:
@ragesoss I think this  ready now

ragesoss commented:
This looks solid. The code is very clean, and from the demo video, it looks like it works as intended.

I think it the UI for the dates panel needs improvement.

The toggle for this feature should probably be just above the weekday selector (in the same section), and ideally it should give a clear visual indication of disabling (or hiding) the weekday and calendar selector when it's active.

Suggestion for the text: "Do not include meeting days or holidays/breaks in the course timeline. (Note: If you don't include meeting days and your course includes vacation weeks, you may need to manually adjust the timeline to account for them.)"

ragesoss commented:
I think so. That would make it more clear that, if they aren't using meeting days, the rest of the inputs on that panel are not applicable.

Formasitchijoh committed:
toggle display of meeting days selector

Formasitchijoh commented:
@ragesoss I updated the feedback

Formasitchijoh commented:
> One minor issue: when enabled, there's an extra horizontal rule that should be removed:
> 
> ![extra line](https://private-user-images.githubusercontent.com/848483/459595299-1bc9b629-3f5b-4edd-8013-f1fe02ea97be.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTA5NzM5NjksIm5iZiI6MTc1MDk3MzY2OSwicGF0aCI6Ii84NDg0ODMvNDU5NTk1Mjk5LTFiYzliNjI5LTNmNWItNGVkZC04MDEzLWYxZmUwMmVhOTdiZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYyNlQyMTM0MjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hM2M3NTIyMGExMjVmYzFkZWQ5NmU4YzdjZDk3Mzg3ZjNkZTc1NmY3ZGU2Y2EwYjE0MDUxOTYzZDg4YzQ2MDBmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.KRxxsED1_YZOF4Z0keuLSYiG0pnK768zrLnskCpEuJo)
> 
So if i am getting you right the input field of the start and end should be flexed horizontal instead of vertical as they currently are to reduce the height of the form
> The new section looks good, but it makes this wizard panel taller than I'd like. I think it might be a good idea to change the layout of the two date entry sections so that the inputs are on the same line.
> 

I realized on editing the course  Dates where the no_meeting_days is true the toggle button does appear is this what you are referring to
> Testing this out locally, I found that the Dates modal doesn't include the 'no meetings' toggle, so if that flag is enabled, the Dates panel is broken. The user sees the Meeting Days and Calendar components, but they don't function correctly. (It looks like it's still possible to click and activate both components, even though they don't highlight whatever you click.)

ragesoss commented:
> the toggle button does appear
 
You mean doesn't appear? Yes, that's what I mean.

Formasitchijoh commented:
yes sorry it does not appear

ragesoss commented:
yeah, that looks good!

Formasitchijoh committed:
improved UI of course dates panel and abstracted switch block into a component

Formasitchijoh committed:
Enabled button on editing course date if it already has course weekdays

ragesoss commented:
great! i think it's ready, but i'll do some hopefully final testing early next week.

ragesoss commented:
@Formasitchijoh please return to this when you get a chance.

Formasitchijoh commented:
> Okay!
> 
> After talking with Helaine... let's make no-dates the default for new courses (and have it represented by the 'off' position of the toggle). Obviously this should not affect existing courses that were created before the feature gets deployed.
> 
> Label: `Specify course meeting days, holidays and breaks?`
> 
> Updated note: `Note: If you don't add meeting days and your course has break weeks, you may need to manually adjust the timeline to account for them.`

Hi @ragesoss  is this the only changes I should fix up here,

ragesoss commented:
yes, plus the merge conflict.

ragesoss commented:
@Formasitchijoh that, plus fixing the conflicts to make it merge-able. Thank you! This is still relevant, and shipping it soon would be great.

Formasitchijoh commented:
> @Formasitchijoh that, plus fixing the conflicts to make it merge-able. Thank you! This is still relevant, and shipping it soon would be great.

Currently testing it out, will  push it soon!

Formasitchijoh committed:
Merge branch 'master' into @feat/no_meeting_days_timeline_support

Formasitchijoh committed:
fix: treat all days as meetings for no-meeting-day courses
 and correct  nextEnabled logic in wizard form panel

- CourseMeetingsManager returned empty day_meetings when weekdays was '0000000', causing all weeks to be blackout weeks and preventing timeline generation for async courses.
- CourseCreationManager now sets no_meeting_days=true on creation so the wizard toggle starts OFF, matching Helaine's UX requirement
- Toggle OFF (no_meeting_days=true) should always enable Next.
Toggle ON requires at least one weekday selected before proceeding.
